src/Storages/ObjectStorage/DataLakes/Iceberg/PartitionPruning.cpp

```diff
@@ -0,0 +1,142 @@
+    {
+        return Iceberg::PartitionTransform::Unsupported;
+    }
+}
+
+// This function is used to convert the value to the start of the corresponding time period (internal ClickHouse representation)
+DateLUTImpl::Values getDateLUTImplValues(Int32 value, Iceberg::PartitionTransform transform)
+{
+    switch (transform)
+    {
+        case Iceberg::PartitionTransform::Year:
+            return DateLUT::instance().lutIndexByYearSinceEpochStartsZeroIndexing(value);
+        case Iceberg::PartitionTransform::Month:
+            return DateLUT::instance().lutIndexByMonthSinceEpochStartsZeroIndexing(static_cast<UInt32>(value));
+        case Iceberg::PartitionTransform::Day:
+            return DateLUT::instance().getValues(static_cast<ExtendedDayNum>(value));
+        case Iceberg::PartitionTransform::Hour:
+        {
+            DateLUTImpl::Values values = DateLUT::instance().getValues(static_cast<ExtendedDayNum>(value / 24));
+            values.date += (value % 24) * 3600;
+            return values;
+        }
+        default:
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unsupported partition transform for get values function: {}", transform);
+    }
+}
+
+// This function is used to convert the value to the start of the corresponding time period (in seconds)
+Int64 getTime(Int32 value, Iceberg::PartitionTransform transform)
+{
+    DateLUTImpl::Values values = getDateLUTImplValues(value, transform);
+    return values.date;
+}
+
+// This function is used to convert the value to the start of the corresponding date period (in days)
+Int16 getDay(Int32 value, Iceberg::PartitionTransform transform)
+{
+    DateLUTImpl::Time got_time = getTime(value, transform);
+    return DateLUT::instance().toDayNum(got_time);
+}
+
+
+Range getPartitionRange(
+    Iceberg::PartitionTransform partition_transform, size_t index, ColumnPtr partition_column, DataTypePtr column_data_type)
+{
+    if (partition_transform == Iceberg::PartitionTransform::Unsupported)
+    {
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unsupported partition transform: {}", partition_transform);
+    }
+    auto column = dynamic_cast<const ColumnNullable *>(partition_column.get())->getNestedColumnPtr();
+    if (partition_transform == Iceberg::PartitionTransform::Identity)
+    {
+        Field entry = (*column.get())[index];
+        return Range{entry, true, entry, true};
+    }
+    const auto [nested_data_type, value] = [&]() -> std::pair<DataTypePtr, Int32>
+    {
+        if (column->getDataType() == TypeIndex::Int32)
+        {
+            const auto * casted_innner_column = assert_cast<const ColumnInt32 *>(column.get());
+            Int32 begin_value = static_cast<Int32>(casted_innner_column->getInt(index));
+            return {dynamic_cast<const DataTypeNullable *>(column_data_type.get())->getNestedType(), begin_value};
+        }
+        else if (column->getDataType() == TypeIndex::Date && (partition_transform == Iceberg::PartitionTransform::Day))
```

Comment:
I just checked and schema inference for value of `Day` transform returns `Date32` type, not `Date`. Did you test it? Or I am missing something.
Also makes sense to write a comment about supported types of partitioned column before this if/else branch

Reply:
Also it's `IColumn::getDataType`, it cannot return `TypeIndex::Date` because we don't have column `ColumnDate`, `Date` type is stored in `ColumnUInt16`, and `Date32` in column `ColumnInt32` so the previous `if` should cover all cases